File: convert_point_mat_to_angle_mat.R
#' Convert a matrix of spatial data into a matrix of angles
#' @param mat the matrix of spatial data, each row corresponds to a point.
#' @description The angles correspond to the angles to the unit vector
#' on each axis.
#' The angles are calculated for each point (row) in mat
#' using \link{convert_point_to_angles_all} which uses
#' \link{convert_point_to_angle} for each column.
#' The angles for each point are then row bound using \link[SparkR]{rbind}
#' into "ang_mat".
#' @return ang_mat
#' @export
mat_to_angle_mat <- function(mat) {
  nc <- ncol(mat)
  nr <- nrow(mat)
  unit_mat <- diag(nc)
  p_list <- lapply(1:nr, convert_point_to_angles_all,
                  mat = mat,
                  unit_mat = unit_mat)
  ang_mat <- Reduce(rbind, p_list)
  ang_mat <- as.matrix(ang_mat)
  row.names(ang_mat) <- row.names(mat)
  colnames(ang_mat) <- colnames(mat)
  return(ang_mat)
}

File: ndimclusteringpy/Python/convert_point_mat_to_angle_mat.py
import numpy as np
import pandas as pd
""" Convert a matrix of spatial data into a matrix of angles

 :param df: the dataframe matrix of spatial data, each row corresponds to a point.

 The angles correspond to the angles to the unit vector
   on each axis.
   The angles are calculated for each point (row) in mat
   using :func:`convert_point_to_angles_all` which uses
   :func:`convert_point_to_angle` for each column.
   The angles for each point are then row bound using pandas concat.
   into "ang_mat".

 :return: ang_mat"""
def mat_to_angle_mat (df) :
  nr, nc = df.shape
  unit_mat = np.identiy(nc)
  p_list = [convert_point_to_angles_all(j,
                  mat = df,
                  unit_mat = unit_mat) 
                  for j in range(1,nr)]
  ang_mat = pd.concat(p_list)
  ang_mat.index = df.index
  ang_mat.columns = df.columns
  return(ang_mat)